src/Plugin/Validation/Constraint/GlobalMessageConstraintValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\stanford_profile_helper\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the UniqueInteger constraint.
+ */
+class GlobalMessageConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+
+    $is_valid = FALSE;
+
+    $resource_fields = [
+      'su_global_msg_label',
+      'su_global_msg_header',
+      'su_global_msg_message',
+      'su_global_msg_link',
+    ];
+
+    if ($value->getEntity()->hasField('su_global_msg_enabled') &&
+    $value->getEntity()->get('su_global_msg_enabled')->getValue()[0]['value']) {
+
+      foreach ($resource_fields as $field) {
+        if (!empty($value->getEntity()->get($field)->getString())) {
+          $is_valid = TRUE;
```

Comment:
if it's true, you could break out of the `foreach`. Tiny performance improvement if you want to.

Reply:
it's a finite loop of 5 elements; breaking out of it early has no measurable performance improvement.

Reply:
it would be extremely tiny improvement. not enough to be of any concern to us. but traversing the 3 different chained methods for each field can add up.